presto-main/src/main/java/com/facebook/presto/tracing/OpenTelemetryTracer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tracing;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.tracing.Tracer;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.facebook.presto.spi.StandardErrorCode.DISTRIBUTED_TRACING_ERROR;
+
+public class OpenTelemetryTracer
+        implements Tracer
+{
+    public static final OpenTelemetry OPEN_TELEMETRY = OpenTelemetryBuilder.build(true);
+    public static final io.opentelemetry.api.trace.Tracer openTelemetryTracer = OPEN_TELEMETRY.getTracer("presto", "1.0.0");
+
+    public final Map<String, Span> spanMap = new ConcurrentHashMap<String, Span>();
+    public final Map<String, Span> recorderSpanMap = new LinkedHashMap<String, Span>();
+
+    public OpenTelemetryTracer()
+    {
+//        openTelemetryTracer = OPEN_TELEMETRY.getTracer("presto", "1.0.0");
+        addPoint("Start tracing");
+    }
+
+    @Override
+    public void addPoint(String annotation)
+    {
+        startBlock(annotation, String.valueOf(System.nanoTime()));
```

Comment:
I can understand why there's a point here, but wondering where this is used in the codebase? May seem to make more sense as just nesting of spans

Reply:
the way this is called, annotation is being passed into `startBlock`'s `blockName` parameter, is this intentional?

Reply:
with the `SimpleTracer`, when it is initialized, a point is added which tracks that time. since there's no equivalent of a point for the otel tracer, i decided it could be a really short block. is that what you're asking? what would nesting of spans be for?

Reply:
yes that is intentional. it will essentially create a span whose name is the annotation (start tracing)

Reply:
for example let's say that there's a point for query completed, one other way this can be designated could be a span from the beginning to the end of the query which would show that information. Wondering depending where this is actually used we can see if we can remove this concept.

Reply:
annotation should probably be used as the same field, and wouldn't use the time as an annotation here;

let's create some way to have some unique placeholder here in the meantime, and maybe push here to see if we can remove this concept

Reply:
addPoint is only used when initializing the tracer and ending the trace, and the points created represent start trace time and end trace time. tracer currently looks like this.

trace start (point) ---- [waiting for prerequisites (span/block)] [dispatching] .... [finished] trace end

for this specific usage, i don't think we can remove add point, because trace start and end can't really be spans.

Reply:
and personally i don't think trace start and end are required (the spans matter more than when the start and end of trace are), so i don't think add point is necessary, but don't think they should be removed either?

Reply:
tldr; i get what you're saying about removing the add point functionality so i'll try to implement a placeholder for it